### PR TITLE
Adding lint files

### LINKED ISSUE #2: Linting
# Description
Regarding linting and code style we should choose which configuration and patterns to use. I'm good with using any current project linting pattern such as Apolo's. What do you guys think?

### PR DESCRIPTION
**Pull request by diogomqbm**
# Description
Closes #2. I added a few extra configs like husky, lint-staged, and a few cypress configs which will be useful in the future when we add cypress.

### CODE REVIEW

File: template/package.json [hunk 1]
```diff
@@ -39,5 +40,28 @@
...
+  "devDependencies": {
+    "eslint": "^7.22.0",
+    "eslint-config-prettier": "^8.1.0",
+    "eslint-plugin-cypress": "^2.11.2",
+    "eslint-plugin-import": "^2.22.1",
+    "eslint-plugin-jsx-a11y": "^6.4.1",
+    "eslint-plugin-prettier": "^3.3.1",
+    "eslint-plugin-react": "^7.22.0",
+    "eslint-plugin-react-hooks": "^4.2.0",
+    "husky": "^4.3.8",
```
Comment (phenrique7) [inline template/package.json:53]: How about bringing the husky [5+ version](https://github.com/typicode/husky/releases/tag/v5.0.0)? The configuration is significantly different for this new version, but I think it is better.

Comment (diogomqbm) [reply template/package.json:53]: I'm getting a strange error and I really don't know why it's happening. Maybe this will take longer than I thought.

Comment (diogomqbm) [reply template/package.json]: I think now I got it.

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npm test
```
Comment (phenrique7) [inline .husky/pre-commit:4]: The `lint` npm script would come here too, wouldn't it?

File: template/package.json [hunk 3]
```diff
@@ -39,5 +41,28 @@
...
+    "eslint-plugin-import": "^2.22.1",
+    "eslint-plugin-jsx-a11y": "^6.4.1",
+    "eslint-plugin-prettier": "^3.3.1",
+    "eslint-plugin-react": "^7.22.0",
+    "eslint-plugin-react-hooks": "^4.2.0",
+    "husky": "^5.1.3",
+    "lint-staged": ">=10",
+    "prettier": "2.2.1"
+  },
+  "husky": {
```
Comment (phenrique7) [inline template/package.json:58]: Is this still necessary?

File: .husky/.gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+_
```
Comment (phenrique7) [inline .husky/.gitignore:1]: Why this or why this file?

File: .husky/pre-commit (hunk 2 above)
Comment (diogomqbm) [reply .husky/pre-commit]: changed

File: template/package.json (hunk 3 above)
Comment (diogomqbm) [reply template/package.json]: removed

File: .husky/.gitignore (hunk 4 above)
Comment (diogomqbm) [reply .husky/.gitignore]: This was created by Husky 5.

**Comment by diogomqbm** [comment]
@phenrique7 done!

**Review by phenrique7 — APPROVED** [review]

**Pull request merged**